pkg/operator/sync.go

```diff
@@ -54,10 +54,10 @@ func (optr *Operator) syncAll(config *OperatorConfig) error {
 	}
 	glog.V(3).Info("Synced up all machine-api-controller components")
 
-	// In addition, if the Provider is BareMetal, then bring up
-	// the baremetal-operator pod
-	if config.BaremetalControllers.BaremetalOperator != "" {
-		if err := optr.syncBaremetalControllers(config); err != nil {
+	// In addition, if the Provider is BareMetal and a provisioning config is available,
+	// then bring up the baremetal-operator pod
+	if baremetalProvisioningConfig, err := getBaremetalProvisioningConfig(optr.dynamicClient, baremetalProvisioningCR); err != nil && apierrors.IsNotFound(err) && config.BaremetalControllers.BaremetalOperator != "" {
```

Comment:
so this would only run `syncBaremetalControllers` when the `baremetalProvisioningConfig` is not found? is that the intention? I thought it was the other way around.

Reply:
There's already a place where you are making this decission -> https://github.com/openshift/machine-api-operator/blob/a52b81d08668851f8faf8db37b38d93fdb15ec7b/pkg/operator/operator.go#L289

Any farther validation should be put in the same place e.g `func usingBareMetal`.

Reply:
I don't like the check I have right now. Please look at the latest version. 

> There's already a place where you are making this decision ->
> 
> https://github.com/openshift/machine-api-operator/blob/a52b81d08668851f8faf8db37b38d93fdb15ec7b/pkg/operator/operator.go#L289
> 
> Any farther validation should be put in the same place e.g `func usingBareMetal`, then you can leave this logic as it is now.

The reasoning for adding the check in the sync is to take care of situations where the Provisioning CR is temporarily unavailable.

Also, there is an additional requirement to provide for a way to opt-in to have a metal3 deployment as part of day-2 operations. So, adding the Provisioning CR at any (later) time would allow the metal3 deployment to proceed.